values.yaml

```diff
@@ -1,402 +1,134 @@
-    # hostname.
+    # coderd.tls.hostSecretName -- The secret to use for TLS.
     hostSecretName: ""
-    # ingress.tls.devurlsHostSecretName -- The secret to use for the
-    # devurls.host hostname.
+    # coderd.tls.devurlsHostSecretName -- The secret to use for DevURL TLS.
     devurlsHostSecretName: ""
-  # ingress.service -- Options used to configure the LoadBalancer Service used
-  # in the default ingress controller.
-  service:
-    # ingress.service.annotations -- Additional annotations to add to the Service
-    # object. For example, to make the ingress spawn an internal load balancer:
-    # annotations:
-    #  cloud.google.com/load-balancer-type: "Internal"
-    annotations: {}
-    # ingress.service.externalTrafficPolicy -- Denotes whether to route external
-    # traffic to node-local or cluster-wide endpoints. "Local" routes external
-    # traffic directly to the correct node, while "Cluster" evenly distributes
-    # traffic across all nodes, but masks source IPs.
-    externalTrafficPolicy: Local
-  # ingress.deployment -- Options used to configure the Deployment of the
-  # default ingress controller.
-  deployment:
-  # ingress.deployment.kind -- How to deploy the built-in ingress. Can be
-  # "Deployment" or "DaemonSet".
-    kind: Deployment
-
-devurls:
-  # devurls.host -- Should be a wildcard hostname to allow matching against
-  # custom-created dev URLs. Leaving as an empty string results in devurls
-  # being disabled. Example: "*.devurls.coder.com".
-  host: ""
-
-# Contains fields related to the Postgres backend. If providing your own
-# instance, a minimum version of Postgres 11 is required with the contrib
-# package installed.
-postgres:
-  # postgres.useDefault -- Deploys an internal Postgres instance alongside the.
-  # platform. We don't recommend running the internal Postgres instance in
-  # production.
-  # If true, all other values are ignored.
-  useDefault: true
-  # postgres.host -- The host of the external postgres instance.
-  host: ""
-  # postgres.port -- The port of the external postgres instance.
-  port: ""
-  # postgres.user -- the user of the external postgres instance.
-  user: ""
-  # postgres.database -- The name of the database that coder will use. It must
-  # exist before Coder is installed.
-  database: ""
-  # postgres.passwordSecret -- The name of an existing secret in the current
-  # namespace with the password to the Postgres instance. The password must be
-  # contained in the secret field `password`. This should be set to an empty
-  # string if the database does not require a password to connect.
-  passwordSecret: ""
-  # postgres.sslMode -- Determines how the connection is made to the database.
-  # The acceptable values are: `disable`, `allow`, `prefer`, `require`,
-  # `verify-ca`, and `verify-full`.
-  sslMode: "require"
-
-# imagePullPolicy -- Sets the policy for pulling a container image across all
-# services.
-imagePullPolicy: Always
```

Comment:
Removing configuration of imagePullPolicy is okay; it can be useful to have this configurable because setting it to Always is required if enforcing image privacy for private repositories (the authorization check happens on image pulls, so using IfNotPresent means that someone can start a container with an image that they shouldn't have access to, in case the cache is warm).

I've heard the argument that people that are concerned about this would enforce it with a web hook, which seems reasonable to me

Reply:
Can we have this set to "Always", always?

Reply:
I think that setting to `Always` means ignoring cache, so it seems like a performance/security tradeoff to me? Maybe we can test it out some more, I'm not sure if `Always` just means it'll check if the image is up-to-date and use the cache if so, or if it would re-download all the image layers

Reply:
`Always` ensures images exist locally with hashes specified in the image manifest.

Reply:
Ah, TIL that `Always` will still use image cache, that makes sense!

It's not very well-linked in the docs, the [page that describes images](https://kubernetes.io/docs/concepts/containers/images/#updating-images) does not link to the [page that describes the details of how imagePullPolicy works](https://kubernetes.io/docs/concepts/configuration/overview/#container-images), though the latter links to the former. Good to know!

> `imagePullPolicy: Always`: every time the kubelet launches a container, the kubelet queries the container image registry to resolve the name to an image digest. If the kubelet has a container image with that exact digest cached locally, the kubelet uses its cached image; otherwise, the kubelet downloads (pulls) the image with the resolved digest, and uses that image to launch the container.